collectors/0/flume_jmx.py

```diff
@@ -12,164 +12,45 @@
-                "org.apache.flume.sink", "",
-                "org.apache.flume.source", "",
-                "Threading", "Count|Time$",       # Number of threads and CPU time.
-                "OperatingSystem", "OpenFile",    # Number of open files.
-                "GarbageCollector", "Collection", # GC runs and time spent GCing.
-                )
-
-        utils.err("Monitoring version: %s Cmd: %s" % (self.version, cmd))
-
-    def run(self):
-        while self._jmx.poll() is None:
-            line = self._jmx.stdout.readline()
-            if len(line) < 4:
-                utils.err("invalid line (too short): %r" % line)
-                continue
-
-            self._process_line(line)
-
-        utils.err("Stopping monitoring version: %s" % self.version)
-
-    def kill_jmx(self):
-        kill(self._jmx)
-
-    def _process_line(self, line):
-        try:
-            timestamp, metric, value, mbean = line.split("\t", 3)
-        except ValueError, e:
-            # Temporary workaround for jmx.jar not printing these lines we
-            # don't care about anyway properly.
-            if "java.lang.String" not in line:
-                utils.err("Can't split line: %r" % line)
-                return
-
-        if metric in java.IGNORED_METRICS:
-            return
-
-        # Sanitize the timestamp.
-        try:
-            timestamp = int(timestamp)
-            if timestamp < time.time() - 600:
-                raise ValueError("timestamp too old: %d" % timestamp)
-            if timestamp < self._prev_timestamp:
-                raise ValueError("timestamp out of order: prev=%d, new=%d"
-                        % (self._prev_timestamp, timestamp))
-        except ValueError, e:
-            utils.err("Invalid timestamp on line: %r -- %s"
-                    % (line, e))
-            return
-
-        self._prev_timestamp = timestamp
-
-        tags = ""
-        # The JMX metrics have per-request-type metrics like so:
-        #   metricNameNumOps
-        #   metricNameMinTime
-        #   metricNameMaxTime
-        #   metricNameAvgTime
-        # Group related metrics together in the same metric name, use tags
-        # to separate the different request types, so we end up with:
-        #   numOps op=metricName
-        #   avgTime op=metricName
-        # etc, which makes it easier to graph things with the TSD.
-        if metric.endswith("MinTime"):  # We don't care about the minimum
-            return                      # time taken by operations.
-        elif metric.endswith("NumOps"):
-            tags = " op=" + metric[:-6]
-            metric = "numOps"
-        elif metric.endswith("AvgTime"):
-            tags = " op=" + metric[:-7]
-            metric = "avgTime"
-        elif metric.endswith("MaxTime"):
-            tags = " op=" + metric[:-7]
-            metric = "maxTime"
+        super(FlumeJmxMonitor, self).__init__(pid, cmd, MONITORED_MBEANS, self.PROCESS_NAME)
```

Comment:
self.PROCESS_NAME -> FlumeJmxMonitor.PROCESS_NAME, is a better convention for class variables